### ISSUE TITLE
UnusedPrivateMembers false-positives

### ISSUE THREAD

**Issue by danh32** [detekt/detekt#2579]
## Expected Behavior
Used private operator functions shouldn't be reported as UnusedPrivateMembers.

## Observed Behavior
Private operator functions are being reported as UnusedPrivateMembers even when they are being used.

## Steps to Reproduce
Foo.kt:
```kotlin
package ...

private operator fun Foo.plus(other: Foo): Foo = Foo(value + other.value)
private operator fun Foo.minus(other: Foo): Foo = Foo(value - other.value)

class Foo(val value: Int) {
    fun double(): Foo = this + this
}

fun main() {
    println(Foo(1).double())
}
```

detekt reports:
```
.../Foo.kt - 10min debt
        UnusedPrivateMember - [plus] at .../Foo.kt:3:1
        UnusedPrivateMember - [minus] at .../Foo.kt:4:1
```

## Context
This seems to be a regression, since it triggered when I'm bumping from 1.7.0 -> 1.7.1. The release notes have a few commits mentioning the `UnusedPrivateMember` rule. I'm guessing this is the relevant change: https://github.com/arturbosch/detekt/pull/2510

It's causing a handful of false-positives in our project.

## Your Environment
* Version of detekt used: 1.7.4
* Version of Gradle used (if applicable): N/A
* Operating System and version: N/A
* Link to your project (if it's a public repository): non-public

**Comment by mcassiano** [comment]
Same rule is also being reported for private extension functions that have overloads. E.g:

```kotlin

private fun A.someMethod(
      param1: B,
      param2: Boolean = true
  ) = someMethod(setOf(param1), param2)

  private fun A.someMethod(
      param1: Set<B>,
      param2: Boolean = true
  ): C { ... }
```

It reports for both functions, even though both overloads are being used.

**Comment by cortinico** [comment]
I've added both reported cases to our tests here #2858 
I wasn't able to reproduce the issue. 

This rule has a "mixed" behavior based on if you have type & symbols resolution enabled. My guess is that you don't have it enabled so that's the cause of those false-positives.
